src/ItemTable.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
+import { useGetItemsQuery } from '../app/services/api';
+import { Button } from 'primereact/button';
+import { Dialog } from 'primereact/dialog';
+import { InputText } from 'primereact/inputtext';
+import { MultiSelect } from 'primereact/multiselect';
+import { DataTable, DataTableSortEvent } from 'primereact/datatable';
+import { Column } from 'primereact/column';
+
+interface Item {
+  id: number;
+  n: string;
+  c: string[];
+}
+
+const ItemTable: React.FC = () => {
+  const { data: items = [] } = useGetItemsQuery(undefined); 
+  const [sortField, setSortField] = useState<string | undefined>(undefined); 

```

Comment:
Is there a reason in some cases you use `| undefined` and in some cases you use `| null` for types?

Also as far as I remember you don't have to explicitly set `undefined` when calling the `useState` hook. Therefore `useState<string>()` would've been enough here.